### PR TITLE
Fix broken following organization

### PR DESCRIPTION
**Pull request by yp05327**
- following organization is broken from #28908
- add login check for the follow button in organization profile page

### CODE REVIEW

**Review by yardenshoham — APPROVED** [review]

**Comment by yardenshoham** [comment]
Sorry for breaking

**Comment by yp05327** [comment]
> Sorry for breaking

Never mind.

**Comment by yp05327** [comment]
@yardenshoham 
~~I can't understand why we not only refresh `follow/unfollow` button, but refresh the whole `profile-avatar-card`~~

I see, we need to refresh the following count in user profile avatar card, but we don't need to refresh it in organization's home page. So only refresh the `follow` button is enough here. I will make a patch for this.

**Comment by yardenshoham** [comment]
So you can remove `hx-target` when templating org

**Comment by yp05327** [comment]
I think this should be in 1.22's milestone, as #28908 will be released in 1.22.

File: routers/web/user/profile.go [hunk 1]
```diff
@@ -318,6 +319,15 @@ func Action(ctx *context.Context) {
...
+	if ctx.ContextUser.IsIndividual() {
+		shared_user.PrepareContextForProfileBigAvatar(ctx)
+		ctx.HTML(http.StatusOK, tplProfileBigAvatar)
+		return
+	} else if ctx.ContextUser.IsOrganization() {
+		ctx.Data["IsFollowing"] = ctx.Doer != nil && user_model.IsFollowing(ctx, ctx.Doer.ID, ctx.ContextUser.ID)
+		ctx.HTML(http.StatusOK, tplFollowUnfollow)
+		return
+	}
+	log.Error("Failed to apply action %q: unsupport context user type: %s", ctx.FormString("action"), ctx.ContextUser.Type)
```
Comment (KN4CK3R) [inline routers/web/user/profile.go:331]: `ctx.ContextUser.Type` is not a `%s`.

File: routers/web/user/profile.go [hunk 2]
```diff
@@ -318,6 +319,15 @@ func Action(ctx *context.Context) {
...
+	ctx.Error(http.StatusBadRequest, fmt.Sprintf("Action %q failed", ctx.FormString("action")))
```
Comment (delvh) [inline routers/web/user/profile.go:332]: Why doesn't the same action apply as for users?
Does the org not use HTMX on that page?
If no, then I'd suggest adding it there as well.
And in the other case, I don't see reasons to deviate from the user behavior…

Comment (yp05327) [reply routers/web/user/profile.go:332]: user profile has the count of followers, that should also be refreshed. But org home page doesn’t.

Comment (delvh) [reply routers/web/user/profile.go:332]: Makes sense, unfortunately.
Ultimately, I think the behavior should be unified, but okay, I can accept that for now.

Comment (yp05327) [reply routers/web/user/profile.go:332]: This also confused me.
You can check the comment: https://github.com/go-gitea/gitea/pull/29005#issuecomment-1920669635

Comment (yardenshoham) [reply routers/web/user/profile.go:332]: I can take a stab at this after this PR is merged

**Review by delvh — APPROVED** [review]
Sorry, forgot the approval yesterday

**Pull request merged**